src/app-layer-parser.c

```diff
@@ -144,6 +144,10 @@ struct AppLayerParserState_ {
     AppLayerDecoderEvents *decoder_events;
 
     FramesContainer *frames;
+
+    uint64_t *tx_id_free_list;
```

Comment:
this is per flow, which will use way too much memory. It should be per thread.

Reply:
Which structure should own it ?
- `FlowWorkerThreadData` ?
- `ThreadVars` ?
- something else ?

Reply:
Perhaps a new `AppLayerSomething...` struct that is a member of `FlowWorkerThreadData`, but we only pass the new struct to the relevant call(s).

Reply:
Ok, I have something in next PR